compiler/fir/resolve/src/org/jetbrains/kotlin/fir/deserialization/FirMemberDeserializer.kt

```diff
@@ -172,13 +173,79 @@ class FirMemberDeserializer(private val c: FirDeserializationContext) {
+            Flags.HAS_ANNOTATIONS.get(flags),
+            Flags.VISIBILITY.get(flags),
+            Flags.MODALITY.get(flags),
+            false, false, false
+        )
+
+        val returnTypeRef = proto.returnType(c.typeTable).toTypeRef(local)
+
+        val getter = if (Flags.HAS_GETTER.get(flags)) {
+            val getterFlags = if (proto.hasGetterFlags()) proto.getterFlags else defaultAccessorFlags
+            val visibility = ProtoEnumFlags.visibility(Flags.VISIBILITY.get(getterFlags))
+            val modality = ProtoEnumFlags.modality(Flags.MODALITY.get(getterFlags))
+            if (Flags.IS_NOT_DEFAULT.get(getterFlags)) {
+                buildPropertyAccessor {
+                    session = c.session
+                    origin = FirDeclarationOrigin.Library
+                    this.returnTypeRef = returnTypeRef
+                    isGetter = true
+                    status = FirDeclarationStatusImpl(visibility, modality)
+                    annotations +=
+                        c.annotationDeserializer.loadPropertyGetterAnnotations(c.containerSource, proto, local.nameResolver, getterFlags)
+                    this.symbol = FirPropertyAccessorSymbol()
+                }
+            } else {
+                FirDefaultPropertyGetter(null, c.session, FirDeclarationOrigin.Library, returnTypeRef, visibility)
+            }
+        } else {
+            null
+        }
+
+        val setter = if (Flags.HAS_SETTER.get(flags)) {
+            val setterFlags = if (proto.hasSetterFlags()) proto.setterFlags else defaultAccessorFlags
+            val visibility = ProtoEnumFlags.visibility(Flags.VISIBILITY.get(setterFlags))
+            val modality = ProtoEnumFlags.modality(Flags.MODALITY.get(setterFlags))
+            if (Flags.IS_NOT_DEFAULT.get(setterFlags)) {
+                buildPropertyAccessor {
+                    session = c.session
+                    origin = FirDeclarationOrigin.Library
+                    this.returnTypeRef = FirImplicitUnitTypeRef(source)
+                    isGetter = false
+                    status = FirDeclarationStatusImpl(visibility, modality)
+                    annotations +=
+                        c.annotationDeserializer.loadPropertySetterAnnotations(c.containerSource, proto, local.nameResolver, setterFlags)
+                    this.symbol = FirPropertyAccessorSymbol()
+                    valueParameters += proto.setterValueParameter.let {
+                        val parameterFlags = if (it.hasFlags()) it.flags else 0
+                        buildValueParameter {
+                            session = c.session
+                            origin = FirDeclarationOrigin.Library
+                            this.returnTypeRef = returnTypeRef
+                            name = if (it.hasName()) c.nameResolver.getName(it.name) else Name.special("<default-setter-parameter>")
+                            this.symbol = FirVariableSymbol(CallableId(FqName.ROOT, name))
```

Comment:
Any reason to give a package name "FqName.ROOT", instead of just FirVariableSymbol(name)?

Reply:
I referred to this: https://github.com/JetBrains/kotlin/blob/master/compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/impl/FirDefaultPropertyAccessor.kt#L112 but to be honest, no idea why. Just tested, and using either doesn't show any differences. Any idea @mglukhikh ?

Btw, note that `FirVariableSymbol(name)` will use `CallableId(name)` creation, which is deprecated.